console-subscriber/src/lib.rs

```diff
@@ -789,87 +789,59 @@ where
-                Some(at)
+                stats.start_poll(now);
             } else if let Some(stats) = exts.get::<Arc<stats::AsyncOpStats>>() {
-                let at = at.unwrap_or_else(Instant::now);
-                stats.start_poll(at);
-                Some(at)
+                stats.start_poll(now);
             // otherwise, is the span a resource? in that case, we also want
             // to enter it, although we don't care about recording poll
             // stats.
             } else if exts.get::<Arc<stats::ResourceStats>>().is_some() {
-                Some(at.unwrap_or_else(Instant::now))
             } else {
-                None
-            }
-        }
+                return;
+            };
 
-        if let Some(span) = cx.span(id) {
-            if let Some(now) = update(&span, None) {
-                if let Some(parent) = span.parent() {
-                    update(&parent, Some(now));
-                }
-                self.current_spans
-                    .get_or_default()
-                    .borrow_mut()
-                    .push(id.clone());
-
-                self.record(|| record::Event::Enter {
-                    id: id.into_u64(),
-                    at: self.base_time.to_system_time(now),
-                });
-            }
+            self.current_spans
+                .get_or_default()
+                .borrow_mut()
+                .push(id.clone());
+
+            self.record(|| record::Event::Enter {
+                id: id.into_u64(),
+                at: self.base_time.to_system_time(now),
+            });
         }
     }
 
     fn on_exit(&self, id: &span::Id, cx: Context<'_, S>) {
-        fn update<S: Subscriber + for<'a> LookupSpan<'a>>(
-            span: &SpanRef<S>,
-            at: Option<Instant>,
-        ) -> Option<Instant> {
+        if let Some(span) = cx.span(id) {
             let exts = span.extensions();
+            let now = Instant::now();
             // if the span we are entering is a task or async op, record the
             // poll stats.
             if let Some(stats) = exts.get::<Arc<stats::TaskStats>>() {
-                let at = at.unwrap_or_else(Instant::now);
-                stats.end_poll(at);
-                Some(at)
+                stats.end_poll(now);
             } else if let Some(stats) = exts.get::<Arc<stats::AsyncOpStats>>() {
-                let at = at.unwrap_or_else(Instant::now);
-                stats.end_poll(at);
-                Some(at)
-                // otherwise, is the span a resource? in that case, we also want
-                // to enter it, although we don't care about recording poll
-                // stats.
+                stats.end_poll(now);
+            // otherwise, is the span a resource? in that case, we also want
+            // to enter it, although we don't care about recording poll
+            // stats.
             } else if exts.get::<Arc<stats::ResourceStats>>().is_some() {
```

Comment:
I think it might make more sense to put this comment inside the (now) empty block following the `else if`

```suggestion
            } else if exts.get::<Arc<stats::ResourceStats>>().is_some() {
                // otherwise, is the span a resource? in that case, we also want
                // to enter it, although we don't care about recording poll
                // stats.
```